File: Definition_and_Pronunciation_Dictionary_App.cs
```CSharp
using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        Dictionary<string, Tuple<string, string>> dictionary = new Dictionary<string, Tuple<string, string>>();
        dictionary.Add("apple", new Tuple<string, string>("A round fruit with red or yellow or green skin", "/ˈæpəl/"));
        dictionary.Add("banana", new Tuple<string, string>("A long curved fruit with yellow skin", "/bəˈnænə/"));

        while (true)
        {
            Console.WriteLine("Enter a word:");
            string word = Console.ReadLine().ToLower();

            if (dictionary.ContainsKey(word))
            {
                Console.WriteLine("Definition: " + dictionary[word].Item1);
                Console.WriteLine("Pronunciation: " + dictionary[word].Item2);
            }
            else
            {
                Console.WriteLine("Word not found in dictionary.");
            }
        }
    }
}
```

File: Definition_and_Pronunciation_Dictionary_App.java
```java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class DictionaryApp {
    public static void main(String[] args) {
        Map<String, String[]> dictionary = new HashMap<>();
        dictionary.put("apple", new String[]{"A round fruit with red or yellow or green skin", "æpl"});
        dictionary.put("banana", new String[]{"A long curved fruit with yellow skin", "bəˈnænə"});
        dictionary.put("cherry", new String[]{"A small round fruit with a single hard pit", "ˈtʃɛri"});

        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("Enter a word (or 'exit' to quit): ");
            String word = scanner.nextLine().toLowerCase();
            if (word.equals("exit")) {
                break;
            }
            if (dictionary.containsKey(word)) {
                System.out.println("Definition: " + dictionary.get(word)[0]);
                System.out.println("Pronunciation: " + dictionary.get(word)[1]);
            } else {
                System.out.println("Word not found in dictionary.");
            }
        }
        scanner.close();
    }
}
```